src/theory/theory_preprocessor.cpp

```diff
@@ -310,9 +310,10 @@ TrustNode TheoryPreprocessor::theoryPreprocess(
         rtfNode = ttfr.getNode();
         registerTrustedRewrite(ttfr, d_tpgRtf.get(), true);
       }
-      // Finish the conversion by rewriting. This is registered as a
-      // post-rewrite, since it is the last step applied for theory atoms.
-      Node retNode = rewriteWithProof(rtfNode, d_tpgRtf.get(), false);
+      // Finish the conversion by rewriting. Notice that we must consider this a
+      // pre-rewrite since we do not recursively register the rewriting steps
+      // of subterms of rtfNode.
```

Comment:
Could you say more about how this change solves the issue of not recursively registering the rewriting steps mentioned?